lib/Fetcher/FeedFetcher.php

```diff
@@ -357,20 +381,20 @@ protected function getFavicon(FeedInterface $feed, string $url)
         $favicon = null;
         // trim the string because authors do funny things
         $feed_logo = $feed->getLogo();
-        
-        if (!is_null($feed_logo)) {
-            $favicon = trim($feed_logo);
-        }
-        
+
         ini_set('user_agent', 'NextCloud-News/1.0');
 
         $base_url = new Net_URL2($url);
         $base_url->setPath("");
         $base_url = $base_url->getNormalizedURL();
-
+        
+        if (!is_null($feed_logo)) {
+            $favicon = trim($feed_logo);
+        }
+        
         // check if feed has a logo entry
         if (is_null($favicon) || $favicon === '') {
-            return $this->faviconFactory->get($base_url);
+            return $this->faviconHelper($url);
```

Comment:
Shouldn't this be `$base_url` still?

Reply:
yea I first wanted to move the base_url generation to that helper function but then noticed it's still needed in the main function. already fixed